packages/datetime/src/dateRangeInput.tsx

```diff
@@ -790,13 +790,18 @@ export class DateRangeInput extends AbstractComponent<IDateRangeInputProps, IDat
             : this.refHandlers.endInputRef;
     }
 
-    private getFormattedDateString = (momentDate: moment.Moment, format?: string) => {
+    private getFormattedDateString = (momentDate: moment.Moment, formatOverride?: string) => {
         if (isMomentNull(momentDate)) {
             return "";
         } else if (!momentDate.isValid()) {
             return this.props.invalidDateMessage;
         } else {
-            return momentDate.format((format != null) ? format : this.props.format);
+            const format = (formatOverride != null) ? formatOverride : this.props.format;
+            if (this.props.locale) {
```

Comment:
I noticed this same code also appeared in #1429. Let's pull this out into a utility function in `dateUtils.ts`:

Maybe:

```tsx
/**
 * Returns a date string in the provided format localized to the provided locale.
 */
export function toLocalizedDateString(momentDate: moment.Moment, format: string, locale: string) {
    const adjustedMomentDate = (locale != null) ? momentDate.locale(locale) : momentDate;
    return adjustedMomentDate.format(format);
}
```